lib/apimatic-core/logger/configurations/api_logging_configuration.rb

```diff
@@ -1,35 +1,35 @@
 module CoreLibrary
   # Represents options for configuring logging behavior.
-  class LoggingConfiguration
-    attr_accessor :logger, :log_level, :request_logging_config, :response_logging_config, :mask_sensitive_headers
+  class ApiLoggingConfiguration
+    attr_reader :logger, :log_level, :request_logging_config, :response_logging_config, :mask_sensitive_headers
 
-    # Initializes a new instance of LoggingConfiguration.
+    # Initializes a new instance of ApiLoggingConfiguration.
     #
     # @param logger [LoggerInterface] The logger to use for logging messages.
     # @param log_level [LogLevel] The log level to determine which messages should be logged.
-    # @param request_logging_config [RequestLoggingConfiguration] Options for logging HTTP requests.
-    # @param response_logging_config [ResponseLoggingConfiguration] Options for logging HTTP responses.
+    # @param request_logging_config [ApiRequestLoggingConfiguration] Options for logging HTTP requests.
+    # @param response_logging_config [ApiResponseLoggingConfiguration] Options for logging HTTP responses.
     # @param mask_sensitive_headers [Boolean] Indicates whether sensitive headers should be masked in logged messages.
     def initialize(
-      logger: ConsoleLogger.new,
-      log_level: ::Logger::INFO,
-      request_logging_config: RequestLoggingConfiguration.new,
-      response_logging_config: ResponseLoggingConfiguration.new,
-      mask_sensitive_headers: true
+      logger: nil,
```

Comment:
Question: Since all parameter default values are nil so can we make them all required?